File: C++/Problems/Problems/problem_561.cpp
#include "problem_561.h"

bool problem_561::test() {
    std::vector nums { 1, 4, 3, 2 };

    int expected = 4;

    auto result = arrayPairSum(nums);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_561::arrayPairSum(std::vector<int> &nums) {
    int n = nums.size();
    std::sort(nums.begin(), nums.end());
    int result = 0;
    for (int i = 0; i < n; i += 2)
    {
        result += nums[i];
    }

    return result;
}

File: Python/Problems/Problems/problem561.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem561(Problem):
    def test(self) -> bool:
        nums = [1, 4, 3, 2]

        expected = 4

        result = self.arrayPairSum(nums)

        print(result)

        return expected == result

    def arrayPairSum(self, nums: List[int]) -> int:
        n = len(nums)
        nums.sort()
        result = 0
        for i in range(0, n, 2):
            result += nums[i]
        return result